src/ir/comp.rs

```diff
@@ -390,26 +399,63 @@ impl CompInfo {
     /// members. This is not ideal, but clang fails to report the size for these
     /// kind of unions, see test/headers/template_union.hpp
     pub fn layout(&self, ctx: &BindgenContext) -> Option<Layout> {
-        use std::cmp;
-
         // We can't do better than clang here, sorry.
         if self.kind == CompKind::Struct {
-            return None;
+            None
+        } else {
+            self.calc_layout(ctx)
         }
+    }
 
-        let mut max_size = 0;
-        let mut max_align = 0;
-        for field in &self.fields {
-            let field_layout = ctx.resolve_type(field.ty)
-                .layout(ctx);
+    /// Compute the layout of this type.
+    pub fn calc_layout(&self, ctx: &BindgenContext) -> Option<Layout> {
```

Comment:
`libclang` will give the struct size after alignment, but sometime we need padding bytes at the end, because Rust doesn't support alignment above 8 bytes.

For example, in the `layout_mbuf` example, we need 24 padding bytes to ensure the `rte_mbuf` has correct size.
```c++
struct rte_mbuf {
	MARKER cacheline0;
       ...
	/** Timesync flags for use with IEEE1588. */
	uint16_t timesync;
} __rte_cache_aligned;
```
```rust
#[repr(C)]
#[derive(Debug, Copy)]
pub struct rte_mbuf {
    pub cacheline0: MARKER,
  ...
    _padding_0: [u64; 3usize],
}
```

`CompInfo::calc_layout` will do the dirty work to find out how many bytes we need padding at the end.